<launch>
    <arg name="sim_time_required" default="false"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <!-- TEMP HERE -->
    <arg name="config_dir" default="$(find hydra)/config/default_robot"/>
    <arg name="use_gt_frame" default="false" doc="use simulator-provided tf frame"/>
    <arg name="use_gt_semantics" default="true" doc="use simulator-provider semantics"/>
    <arg name="use_static_tfs" default="true" doc="publish static tfs from file"/>
    <arg name="color_mesh_by_label" default="false" doc="display mesh colored by semantic label"/>
    <arg name="use_oriented_bounding_boxes" default="false"/>
    <arg name="use_2d_places" default="true"/>
    <arg name="use_single_channel_label_image" default="false"/>

    <!-- NOTE: this is to align with CMU dev low level navigation -->
    <arg name="robot_frame" default="base_link" doc="robot body tf frame, used as the central reference point for robot components"/>
    <arg name="odom_frame" default="odom" doc="odometry (map) frame, used as the starting point for navigation and localization"/>
    <arg name="sensor_frame" default="left_cam" doc="primary sensor frame, used for initial state estimations and primary sensor data"/>

    <!-- ADAM MODIFED HERE -->
    <arg name="use_zmq_interface" default="false"/>
    <!-- <arg name="zmq_ip" default="127.0.0.1"/>
    <arg name="zmq_send_url" default="tcp://$(arg zmq_ip):1641"/>
    <arg name="zmq_recv_url" default="tcp://$(arg zmq_ip):1642"/> -->

    <arg name="start_pose_graph_viewer" default="true"/>
    <arg name="world_frame" default="world"/>

    <!-- semantics -->
    <!-- TODO: Correct this! -->
    <arg name="labelspace_name" default="$(eval 'uhumans2_office' if arg('use_gt_semantics') else 'ade20k_mp3d')" doc="semantic label space"/>
    <arg name="semantic_map_dir" default="$(find hydra_ros)/config/color" if="$(arg use_gt_semantics)"/>
    <arg name="semantic_map_dir" default="$(find semantic_inference)/config/colors" unless="$(arg use_gt_semantics)"/>
    <arg name="semantic_map_path" default="$(arg semantic_map_dir)/$(arg labelspace_name).csv"/>
    <arg name="semantic_label_remap_filepath" default="$(find hydra)/config/label_remaps/uhumans2_office.yaml" if="$(arg use_single_channel_label_image)"/>

    <!-- ADAM: changed to match habitat frontend render topics -->
    <!-- <arg name="rgb_topic" default="/camera/color/image_raw"/>
    <arg name="rgb_info_topic" default="/camera/color/camera_info"/>
    <arg name="depth_topic" default="/camera/aligned_depth_to_color/image_raw"/>
    <arg name="depth_info_topic" default="/camera/depth/camera_infos"/>
    <arg name="label_topic" default="/camera/color/image_raw" unless="$(arg use_gt_semantics)"/> -->

    <!-- <arg name="pointcloud_topic" default="semantic_pointcloud"/> -->
    <arg name="rgb_topic" default="/camera/color/image_raw"/>
    <arg name="rgb_info_topic" default="/camera/color/camera_info"/>
    <arg name="depth_info_topic" default="/camera/aligned_depth_to_color/camera_info"/>
    <arg name="depth_topic" default="/camera/aligned_depth_to_color/image_raw"/>
    <arg name="label_topic" default="/camera/color/image_raw"/>


    <arg name="debug" default="false"/>

    <!-- NOTE USED???? -->
    <!-- <arg name="prerecorded_topic" default="/oneformer/labels/image_raw"/> -->

    <!-- <include file="$(find hydra_ros)/launch/static_tfs/D435i.xml" if="$(arg use_static_tfs)"/> -->

    <!-- TODO: Maybe this is not needed -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="fake_base_link_tf" args="0 0 0 0 0 0 1 map base_link" /> -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_odom" args="0 0 0 -0.5 -0.5 0.5 0.5 odom base_link" /> -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="fake_map_tf" args="0 0 0 0 0 0 1 odom world" />
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="fake_tf_world" args="0 0 0 0 0 0 1 world map" /> -->

    <include file="$(find semantic_inference_ros)/launch/semantic_inference.launch"
             pass_all_args="true"
             unless="$(arg use_gt_semantics)">
        <arg name="enable_pointcloud" value="false"/>
    </include>

    <include file="$(find hydra_ros)/launch/hydra.launch" pass_all_args="true">
        <arg name="dataset_name" default="d435i"/>
        <arg name="rviz_file" default="uhumans2.rviz"/>
        <arg name="kimera_sensor_filepath" default="$(find kimera_vio)/params/RealSenseIR/LeftCameraParams.yaml"/>
    </include>

</launch>
